src/main/java/frc/robot/motors/SparkMaxWrapper.java

```diff
@@ -0,0 +1,106 @@
+package frc.robot.motors;
+
+import com.revrobotics.sim.SparkMaxSim;
+import com.revrobotics.spark.SparkMax;
+
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.RobotController;
+import frc.robot.Robot;
+
+/**
+ * Spark max wrapper class, using SparkMaxSim.
+ */
+public class SparkMaxWrapper extends SparkMax implements LoggedMotor {
+
+	// Components
+	private final DCMotor motorType;
+	private final SparkMaxSim motorSim;
+
+	public static final double LOOP_PERIOD_MS = 0.020;
+
+	private double targetVelocity = 0;
+
+	/**
+	 * Constructor for CAN ID and motor type.
+	 * @param deviceId the CAN ID of the motor
+	 * @param type the type of motor
+	 */
+	public SparkMaxWrapper(int deviceId, MotorType type) {
+		// Initialize motor
+		super(deviceId, type);
+		init();
+
+		// Create sim instance
+		motorType = DCMotor.getNEO(1);
```

Comment:
If we're assuming a NEO motor, then the MotorType should always be brushless. Also is there some way we can model NEO550 motors with this class?